src/editor/Store/VimStoreConnector.re

```diff
@@ -154,28 +154,35 @@ let start = (getState: unit => Model.State.t, getClipboardText) => {
       Log.info("Vim.Window.onMovement");
       let currentState = getState();
 
-      let v =
-        switch (movementType) {
-        | FullLeft
-        | OneLeft => Model.WindowManager.moveLeft(currentState.windowManager)
-        | FullRight
-        | OneRight =>
-          Model.WindowManager.moveRight(currentState.windowManager)
-        | FullDown
-        | OneDown => Model.WindowManager.moveDown(currentState.windowManager)
-        | FullUp
-        | OneUp => Model.WindowManager.moveUp(currentState.windowManager)
-        | _ => currentState.windowManager.activeWindowId
+      let move = (moveFunc, state) => {
+        let windowId = moveFunc(currentState.windowManager);
+        let maybeEditorGroupId =
+          Model.WindowTree.getEditorGroupIdFromSplitId(
+            windowId,
+            currentState.windowManager.windowTree,
+          );
+
+        switch (maybeEditorGroupId) {
+        | Some(editorGroupId) =>
+          dispatch(Model.Actions.WindowSetActive(windowId, editorGroupId))
+        | None => ()
         };
+      };
 
-      let editorId =
-        Model.WindowTree.getEditorGroupIdFromSplitId(
-          v,
-          currentState.windowManager.windowTree,
-        );
-      switch (editorId) {
-      | Some(ed) => dispatch(Model.Actions.WindowSetActive(v, ed))
-      | None => ()
+      switch (movementType) {
+      | FullLeft
+      | OneLeft => move(Model.WindowManager.moveLeft, currentState)
```

Comment:
I was not entirely sure what's best practice here. I don't technically have to pass `currentState` to `move` due to lexical scope. `currentState` is defined in the outer scope. However, passing `currentState` makes the dependency obvious.

Thoughts?

Reply:
You are not using `state` but `currentState` in your move function, so now the argument makes no sense at all.
I probably wouldn't pass the state in, but not sure what the best practices are for this.

Reply:
Oh, your're right. Nice catch! 🙂 Well, given that the `move` function do use the `state` parameter, does it make sense?

Reply:
I don't think it matters much, and think it's more of a style thing.